docs/src/examples/components/Popup/Variations/PopupExamplePosition.shorthand.tsx

```diff
@@ -1,145 +1,67 @@
 import React from 'react'
 import { Button, Grid, Popup } from '@stardust-ui/react'
 
+class PopupArrowExample extends React.Component<any, any> {
+  state = { popupOpen: false }
+
+  togglePopupState = () => {
+    this.setState(prev => ({ popupOpen: !prev.popupOpen }))
```

Comment:
because `setState` applies changes asynchronously, so the way you are suggesting won't be reliable in a sense that we might read a 'stale' value for `popupOpen` prop. Please, take a look here: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

Reply:
oh, disregard my comment

I knew about that, but I think it's relevant for complex state updates from different entry points that might be batched together; however, I agree it's better like u suggest; first time I read that part I misunderstood some of it, thx for reminder ;)